crates/driver/src/infra/config/cli.rs

```diff
@@ -143,6 +181,77 @@ impl std::fmt::Display for ContractAddresses {
     }
 }
 
-pub fn hex_address(value: &str) -> eth::H160 {
-    eth::H160::from_str(value).unwrap()
+/// Parameters related to settlement submission.
+#[derive(Debug, clap::Parser)]
+pub struct Submission {
+    /// How to calculate gas prices. Multiple approaches are used in sequence if
+    /// the previous one fails.
+    /// eth-gas-station: supports mainnet.
+    /// gas-now: supports mainnet.
+    /// gnosis-safe: supports mainnet and goerli.
+    /// web3: supports every network.
+    /// native: supports every network.
+    #[clap(
+        long,
+        env,
+        default_value = "web3",
+        value_enum,
+        use_value_delimiter = true
+    )]
+    pub submission_gas_price_calculation: Vec<GasPriceCalculation>,
+
+    /// Additional tip in percentage of max_fee_per_gas we are willing to give
+    /// to miners above regular gas price estimation. Expects a floating point
+    /// value between 0 and 1.
+    #[clap(long, env, default_value = "0.05")]
+    pub submission_additional_tip_percentage: f64,
+
+    /// The maximum gas price in Gwei the solver is willing to pay in a
+    /// settlement.
+    #[clap(long, env, default_value = "1500")]
+    pub submission_gas_price_cap: f64,
+
+    /// The target confirmation time for settlement transactions used
+    /// to estimate gas price. Specified in seconds.
+    #[clap(long, env, default_value = "30")]
+    pub submission_target_confirm_time_secs: u64,
+
+    /// Amount of time to wait before retrying to submit the tx to
+    /// the ethereum network. Specified in seconds.
+    #[clap(long, env, default_value = "2")]
+    pub submission_retry_interval_secs: u64,
+
+    /// Don't submit transactions with high revert risk (i.e. transactions that
+    /// interact with on-chain AMMs) to the public mempool. This can be
+    /// enabled to avoid MEV when private transaction submission strategies
+    /// are available.
+    #[clap(long, env)]
+    pub submission_disable_high_risk_public_mempool_transactions: bool,
+
+    /// The maximum time to spend trying to settle a transaction
+    /// through the Ethereum network before going back to solving. Specified in
+    /// seconds.
+    #[clap(long, env, default_value = "120")]
+    pub submission_max_confirm_time_secs: u64,
+
+    /// Maximum additional tip in Gwei that we are willing to give to flashbots
+    /// above regular gas price estimation.
+    #[clap(long, env, default_value = "3")]
+    pub submission_max_additional_flashbots_tip: f64,
+}
+
+#[derive(Debug, Clone, Copy, clap::ValueEnum)]
+pub enum Mempool {
+    Public,
+    Flashbots,
+}
```

Comment:
Not sure if `Mempool` is the right name (with regard to `Flashbots` and `Gelato`).
Maybe `Relayer`? But this would also make the `Public` variant fit less.

Reply:
I think for `Flashbots` it is (it is just a private mempool). Gelato is a relayer, and isn't a `Mempool`.

Reply:
Gelato submits a "call" to their "relay network". A relayer picks up the call and is responsible for crafting and executing the transaction on the blockchain (the relayer _probably_ uses some mempool, but this is orthogonal to the "call" we submit to their "relay network").

Reply:
> Do we even want to support Gelato?

Quite a bit of [work](https://github.com/cowprotocol/services/blob/main/crates/shared/src/gelato_api.rs) was done to work toward that. The nice thing about Gelato is that they would take care of all the trickiness involved in submitting stuff.

Reply:
> Do we even want to support Gelato?

Yes, eventually we want to support even other relayers (Infura ITX for example).

I do think it fits nicely with the way things are currently organized, so I wouldn't worry too much about optimizing the current abstractions for it until they are needed.

I think `Mempool` or `Txpool` makes sense as a name for now.